src/connect.ts

```diff
@@ -0,0 +1,103 @@
+import { Coin, DirectSecp256k1HdWallet } from "@cosmjs/proto-signing";
+import { Block, calculateFee, GasPrice } from "@cosmjs/stargate";
+import {SigningStargateClient, StargateClient } from "@cosmjs/stargate";
+import { ArchwayClient, ArchwayTxFilter, ArchwayTXSearch } from "./query";
+import { SigningCosmWasmClient, CosmWasmClient } from "@cosmjs/cosmwasm-stargate";
+import {uploadContract} from "./wasm"
+// Wrapper for connecting to RPC client from COSM JS
+// Can either connect to testnet or mainnet
+// TODO: add mainnet
+
+
+// Interface that holds the Signing Client and Wallet Account
+export class ArchwaySigningClient {
+    wallet: DirectSecp256k1HdWallet;
+    client: SigningCosmWasmClient;
+    public uploadContract = uploadContract;
+
+    constructor(wallet:DirectSecp256k1HdWallet, client: SigningCosmWasmClient) {
+        this.client = client;
+        this.wallet = wallet;
+    }
+
+    async getBalance(): Promise<Coin> {
+        const address = await extractAddress(this.wallet);
+        const balance = await this.client.getBalance(address, "uconst");
+        return balance
+    };
+    sendTokens(archway_recipient_address: string, amount: Coin[]): void {};
+    delegateTokens(archway_validator_address: string, amount: Coin): void {};
+
+}
+
+
+
+
+
+
+// Generate a wallet structure randomly 
+export async function generateWallet() {
+    const wallet: DirectSecp256k1HdWallet = await DirectSecp256k1HdWallet.generate(15, {prefix: "archway"});
```

Comment:
when I mention hardcoded, I refer also to this `prefix`, rpc url, etc

Reply:
When there is a mainnet there would be an option to connect to either mainnet or testnet.